notebooks/ch12/linear_case_study.md

```diff
@@ -0,0 +1,1205 @@
+      <td>0</td>
+      <td>0</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 13 columns</p>
+</div>
+
+
+
+We should also add a column of biases in order to have a constant term in our model.
+
+
+```python
+donkeys_train = donkeys_train.assign(bias=1)
+```
+
+
+```python
+# HIDDEN
+donkeys_train = donkeys_train.reindex_axis(['bias'] + list(donkeys_train.columns[:-1]), axis=1)
+
+```
+
+
+```python
+donkeys_train.head()
+```
+
+
+
+
+<div>
+<style>
+    .dataframe thead tr:only-child th {
+        text-align: right;
+    }
+
+    .dataframe thead th {
+        text-align: left;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>bias</th>
+      <th>Length</th>
+      <th>Girth</th>
+      <th>...</th>
+      <th>Age_2-5</th>
+      <th>Age_&lt;2</th>
+      <th>Age_&gt;20</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>465</th>
+      <td>1</td>
+      <td>98</td>
+      <td>113</td>
+      <td>...</td>
+      <td>1</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>233</th>
+      <td>1</td>
+      <td>101</td>
+      <td>119</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>450</th>
+      <td>1</td>
+      <td>106</td>
+      <td>125</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>453</th>
+      <td>1</td>
+      <td>93</td>
+      <td>120</td>
+      <td>...</td>
+      <td>1</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>452</th>
+      <td>1</td>
+      <td>98</td>
+      <td>120</td>
+      <td>...</td>
+      <td>0</td>
+      <td>0</td>
+      <td>0</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 14 columns</p>
+</div>
+
+
+
+## Multiple Linear Regression Model
+
+We are finally ready to fit our model!
+
+Our model looks like this:
+
+$$
+f_\hat{\theta} (x) = \hat{\theta_0} + \hat{\theta_1} (Length) + \hat{\theta_2} (Girth) + \hat{\theta_3} (Height) + ... +  \hat{\theta_{13}} (Age_>20)
+$$
+
+Remember, the $\hat{\theta_0}$ is accounted for due to the bias column in our dataframe.
+
+Here are the functions we defined earlier in the multiple linear regression section, which we will use again:
+
+
+```python
+def linear_model(thetas, X):
+    '''Returns predictions by a linear model on x_vals.'''
+    return X @ thetas
+
+def mse_cost(thetas, X, y):
+    return np.mean((y - linear_model(thetas, X)) ** 2)
+
+def grad_mse_cost(thetas, X, y):
+    n = len(X)
+    return -2 / n * (X.T @ y  - X.T @ X @ thetas)
+```
+
+In order to use the above functions, we need `X`, and `y`. These can both be obtained from our data frames. Remember that `X` and `y` have to be numpy matrices in order to be able to multiply them with `@` notation.
+
+`X` consists of all columns of the data frame `donkeys_train`.
+
+
+```python
```

Comment:
I think we can just use donkeys_train and y_train; I think it's more clear that way

Reply:
hmm i kind of like keeping the dataframe and the numpy array separate though